### ISSUE TITLE
Invalid yum/apt repositories generated

### ISSUE THREAD

**Issue by tequeter** [voxpupuli/puppet-gitlab#273]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5
- Distribution: RHEL7
- Module version: master branch

## How to reproduce (e.g Puppet code you use)

```puppet
class { 'gitlab':
}
```

## What are you seeing

There is an incomplete yum repos `gitlab_official_ce` with just the baseurl in it, and a `gitlab_official_` yum repos with all `ce` strings missing (name and URL). Yum cannot use either of those and the Gitlab package cannot be installed.

The apt sources suffer from the same bug.

## What behaviour did you expect instead

A valid `gitlab_official_ce` definition usable by yum/apt.

## Any additional information you'd like to impart

See test case in tequeter@ea67bd4460cd2e6dce99119bf0e0d40a8b811d55.

**Comment by LongLiveCHIEF** [comment]
Thanks for opening this. If nothing else, it helps our changelog generator show bug fixes in the correct format. There's definitely a problem with this code, and as you pointed out in #271 hiera wasn't meant to work the way I wrote this bit of code for.

I left a few other comments in the PR, and I'm taking another look at what I was planning things to look like when I removed the deprecated v2 class params in light of some code you submitted in your PR.
